### PR TITLE
fix #298438: Correctly import notes without beams from MusicXML.

### PR DESCRIPTION
**Pull request by matangover**
Resolves: [298438](https://musescore.org/en/node/298438)

Previously, notes without beams were always being set to auto beaming when imported from MusicXML. Now, notes without beams are correctly imported without beams.

The old automatic beaming behavior is retained only if:
1. The score does not specify `<supports element="beam" type="yes">`, AND
2. The score does not contain any `<beam>` elements.

I've tried to add tests in tst_mxml_io.cpp but it seems that this test is already failing as it is. I have tested manually using several files: [beam_tests.zip](https://github.com/musescore/MuseScore/files/5791367/beam_tests.zip)

Related: issue [100901](https://musescore.org/en/node/100901) and PR [3221](https://github.com/musescore/MuseScore/pull/3221). It seems that this PR is the one that caused this import bug. I have verified that the XML file from the original issue still works correctly after the current PR.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by matangover** [comment]
OK apparently tst_mxml_io does pass before this PR and I just couldn't get it to run correctly on my machine, let me correct that

**Comment by matangover** [comment]
OK, done. I've fixed the failing test, and added three tests for beam importing.

I've also fixed a subtle bug in the tests where if a single test was run in isolation, it would fail (test result depended on order of running tests), because `PREF_EXPORT_MUSICXML_EXPORTLAYOUT` was only being set sometimes.

**Comment by RobFog** [comment]
@lvinken Could you have a look?

**Comment by lvinken** [comment]
This change also sets beam mode no on the quarter notes. Does not really seem to harm, but is superfluous.

**Review by lvinken — COMMENTED** [review]
One more question: does this PR affect the new test file in PR7213 ? If so, it may be more practical to provide a combined PR.

Apart from that and my previous remarks, this seems OK to me, thanks for fixing it.

File: importexport/musicxml/importmxmlpass2.cpp [hunk 1]
```diff
@@ -1810,10 +1810,14 @@ static void handleBeamAndStemDir(ChordRest* cr, const Beam::Mode bm, const Direc
...
                   }
             }
-      // if no beam, set stem direction on chord itself and set beam to auto
+      // if no beam, set stem direction on chord itself
       if (!beam) {
             static_cast<Chord*>(cr)->setStemDirection(sd);
-            cr->setBeamMode(Beam::Mode::AUTO);
+            // set beam to none if score has beaming information, otherwise to auto
+            if (hasBeamingInfo)
+                  cr->setBeamMode(Beam::Mode::NONE);
```
[the remark this answers is not in the document]
Comment (matangover) [reply importexport/musicxml/importmxmlpass2.cpp:1818]: Good point. I'll do that

**Comment by matangover** [comment]
The test file from the other PR is deliberately not affected, they are independent

**Comment by lvinken** [comment]
OK, no further comments from my side.

File: importexport/musicxml/importmxmlpass2.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (matangover) [reply importexport/musicxml/importmxmlpass2.cpp:1823]: done

**Comment by matangover** [comment]
Thanks for the review. I fixed the comment, the PR ready to merge as far as I'm concerned.

**Comment by vpereverzev** [comment]
Looking for rebase

**Comment by matangover** [comment]
Rebase done

**Pull request merged**